### PR TITLE
improve consult line experience

### PR DESCRIPTION
**Pull request by s-kostyaev**
Add optional arg `initial` to consult line. Implement `consult-line-symbol-at-point` & `consult-line-from-isearch`. See #16

### CODE REVIEW

File: consult.el [hunk 1]
```diff
@@ -547,10 +547,10 @@ WIDTH is the line number width."
     (cons default-cand (nreverse candidates))))
 
 ;;;###autoload
-(defun consult-line ()
+(defun consult-line (&optional initial)
   "Search for a matching line and jump to the line beginning.
 The default candidate is a non-empty line closest to point.
-This command obeys narrowing."
+This command obeys narrowing. Optionaly you can provide INITIAL input."
```
Comment (minad) [inline consult.el:553]: typo optionally

File: consult.el [hunk 2]
```diff
@@ -562,8 +562,21 @@ This command obeys narrowing."
                       :history 'consult-line-history
                       :lookup (lambda (candidates x) (cdr (assoc x candidates)))
                       :default (car candidates)
+		      :initial initial
```
Comment (minad) [inline consult.el:565]: formatting

**Review by minad — COMMENTED** [review]
Two minor nitpicks. And please document the commands in the readme too. Looks good besides that!

File: consult.el [hunk 3]
```diff
@@ -562,8 +562,21 @@ This command obeys narrowing."
...
+(defun consult-line-symbol-at-point ()
+  "Search for a symbol at point."
+  (interactive)
+  (consult-line (thing-at-point 'symbol)))
+
+;;;###autoload
+(defun consult-line-from-isearch ()
+  "Search by lines from isearch string."
+  (interactive)
+  (consult-line isearch-string))
```
Comment (minad) [inline consult.el:578]: swiper seems to have something like this: 

~~~ elisp
(if isearch-regexp
                   isearch-string
                 (regexp-quote isearch-string))))
~~~

Add something like that too?

**Comment by s-kostyaev** [comment]
@minad Thanks for review. Some fixes provided.

File: consult.el (hunk 3 above)
Comment (s-kostyaev) [reply consult.el]: Sure.

File: consult.el (hunk 2 above)
Comment (s-kostyaev) [reply consult.el]: fixed.

File: consult.el (hunk 1 above)
Comment (s-kostyaev) [reply consult.el]: fixed.

**Comment by minad** [comment]
Merged after squashing. Thank you.

**Pull request closed without merging**